cmd/src/main.go

```diff
@@ -178,3 +244,65 @@ func readConfig() (*config, error) {
 func cleanEndpoint(urlStr string) string {
 	return strings.TrimSuffix(urlStr, "/")
 }
+
+// isValidUnixSocket checks if the given path is a valid Unix socket.
+//
+// Parameters:
+//   - path: A string representing the file path to check.
+//
+// Returns:
+//   - bool: true if the path is a valid Unix socket, false otherwise.
+//   - error: nil if the check was successful, or an error if an unexpected issue occurred.
+//
+// The function attempts to establish a connection to the Unix socket at the given path.
+// If the connection succeeds, it's considered a valid Unix socket.
+// If the file doesn't exist, it returns false without an error.
+// For any other errors, it returns false and the encountered error.
+func isValidUnixSocket(path string) (bool, error) {
+	conn, err := net.Dial("unix", path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, errors.Newf("Not a UNIX Domain Socket: %v: %v", path, err)
```

Comment:
```suggestion
		return false, errors.Newf("Not a UNIX Domain Socket: %v: %w", path, err)
```

Reply:
Whoops; meant to use `w` instead of `v`! Thanks for catching that.